addons/website_sale/static/src/js/sale_variant_mixin.js

```diff
@@ -548,19 +521,12 @@ var VariantMixin = {
             $compare_price.removeClass("d-none");
         }
 
-        var rootComponentSelectors = [
-            'tr.js_product',
-            '.oe_website_sale',
-        ];
-
-        // update images only when changing product
+        // update images & tags only when changing product
         // or when either ids are 'false', meaning dynamic products.
         // Dynamic products don't have images BUT they may have invalid
         // combinations that need to disable the image.
```

Comment:
I think it still is :thinking:

Reply:
Hmm, maybe you only changed the way you check this, but I don't think we still have the equivalent of `!this.last_product_id`

Reply:
`last_product_id` was set to the product (potentially empty) of the last combination.
The only case where there is a change is the first rpc to `get_combination_info`, when the page opens.

Two possibilities:
1) the product returned by that first rpc is the one provided (found in the rendered page content)
Previously, the if was entered, but it was meaningless, since the images and tags on the page are already the right ones.  --> Different behavior, but it was useless before to rerender the image and tags directly.
2) the product is different (very low probability since there is no cache on the page, it would mean that the product configuration was changed between the page rendering and the first rpc)
We correctly enter the if since there was a product change --> Same behavior as before.